shared/naturalcrit/splitPane/splitPane.jsx

```diff
@@ -42,6 +43,22 @@ const SplitPane = createClass({
 			});
 		}
 		window.addEventListener('resize', this.handleWindowResize);
+
+
+		// This lives here instead of in the initial render because you cannot touch localStorage until the componant mounts.
+		const loadLiveScroll = window.localStorage.getItem('liveScroll') === 'true';
+		this.setState({
+			liveScroll : loadLiveScroll
+		});
+		const toggle = document.getElementById('scrollToggle');
+		const toggleDiv = document.getElementById('scrollToggleDiv');
+		if(loadLiveScroll) {
+			toggle.className = 'fas fa-lock';
+			toggleDiv.className = 'arrow lock';
+		} else {
+			toggle.className = 'fas fa-unlock';
+			toggleDiv.className = 'arrow unlock';
+		}
```

Comment:
In React, we don't want to modify the DOM directly like this. This can get React's shadow copy of the DOM out of sync and unable to detect changes.

Instead of managing the Class changes here, you would rely on your `liveScroll` state and change the `render` output  below based on the state.

So the flow should be:

Click button -> change the component state -> Render the updated component based on the state value.